Client/Frontend/Home/HistoryHighlights/HistoryHighlightsManager.swift

```diff
@@ -0,0 +1,125 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Foundation
+import Shared
+import Storage
+import MozillaAppServices
+
+protocol HighlightItem {}
+
+extension ASGroup: HighlightItem {}
+extension HistoryHighlight: HighlightItem {}
+
+extension HistoryHighlight {
+    var urlFromString: URL? {
+        return URL(string: url)
+    }
+}
+
+class HistoryHighlightsManager {
+
+    // MARK: - Variables
+
+    // These variables are defined by PM and Design and will be tweaked based on
+    // their requirements or input.
+    private static let defaultViewTimeWeight = 10.0
+    private static let defaultFrequencyWeight = 4.0
+    private static let defaultHighlightCount = 9
+    private static var shouldGroupHighlights = false
+
+    // MARK: - Public interface
+
+    /// Get HistoryHighlitght from A~S then filter open tabs from the history highlight result there should not be duplicated
+    /// after apply group logic if `shouldGroupHighlights` is set to true and finally collate indidual HistoryHighlight with `ASGroup<HistoryHighlight>`
+    /// to return the top nine results alternating betwen them.
```

Comment:
```suggestion
    /// Fetches HistoryHighlight from A~S, and then filters currently open
    /// tabs against history highlights in order to avoid duplicated items. Then,
    /// if `shouldGroupHighlights` is set to true, applies group logic and finally, 
    /// collates indivdual HistoryHighlight with `ASGroup<HistoryHighlight>`
    /// to return the top nine results alternating betwen them.
    ///
```